hikari/api/special_endpoints.py

```diff
@@ -1838,5 +1881,63 @@ def add_select_menu(self: _T, custom_id: str, /) -> SelectMenuBuilder[_T]:
         SelectMenuBuilder[Self]
             Select menu builder object.
             `SelectMenuBuilder.add_to_container` should be called to finalise the
-            button.
+            component.
+        """
+
+
+class ModalActionRowBuilder(ComponentBuilder, abc.ABC):
```

Comment:
Considering we will always need an ActionRow, why dont we create "Action Row mixins" that specify what components can be added?

This may take a bit of a rework of how ComponentBuilders function (most probably insanely breaking), but instead of the different `add_x`, we could leverage `add_component` and have an `ActionRow` to which you specify what components are allowed to be added to it, providing a different way of adding them that isnt chaining calls.

I wasnt able to comment on the discussion that went down when `ModalActionRowBuiler` was suggested, but ill make my comment now:

I see how this is needed. Even if we abstract over the API, for REST Bots it is necessary, as Discord gives 0 feedback on why an interaction failed (they cant because we are just responding to their request to get a respond to a specific interaction) and adding components that shouldn't be added to an action row may be confusing. But having separate ActionRow builders might not be the way to ensure this and makes it a bit hard to work with

Reply:
Can we like, not do that in this pr? It's already bloated enough. Probably should be a completely separate PR tbh.

Reply:
This feels like an abstration that isn't required in hikari in my opinion.